### ISSUE TITLE
When in '_section', use properties outside this '_section'

### ISSUE THREAD

**Issue by hiulit** [MathiasPaumgarten/grunt-bake#101]
I have an `/index.html`

```
<!--(bake /includes/header.html _section="pages.home")-->
    <section class="section section--1">Section 1</section>
<!--(bake /includes/footer.html)-->
```

and a `/pages/example/index.html`

```
<!--(bake /includes/header.html _section="pages.example")-->
    <section class="section section--1">Example</section>
<!--(bake /includes/footer.html)-->
```
The `header.html`

```
<!DOCTYPE html>
<html lang="en">
<head>
    <!--(bake /includes/metas.html)-->
    <link href="/styles/main.css" rel="stylesheet">
</head>
```

And the `metas.html`

```
<meta charset="UTF-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>{{metas.title}}</title>
<meta name="description" content="{{metas.description}}">
<meta name="author" content="{{author}}">
```

And finally, this `final.json`

```
{
  "metas": {
    "author": "Sir Arthur Conan Doyle",
    "ga": "U-xxx",
    "twitter": "@example",
    "url": "http://www.example.com",
    "share-image": "path/to/share-image.jpg"
  },
  "pages": {
    "home": {
      "id": "home",
      "metas": {
        "description": "Description must be 160 characters maximum and different of each other.",
        "title": "Title must be between 50-60 characters maximum and different of each other."
      }
    },
    "example": {
      "id": "example",
      "metas": {
        "description": "Description must be 160 characters maximum and different of each other.",
        "title": "Title must be between 50-60 characters maximum and different of each other."
      }
    }
  }
}
```

What I would like to do is something like `{{../../metas.author}}` to get the author property from `metas.author`. 
Is it posible? To get a property outside the current `_section`?

**Comment by MathiasPaumgarten** [comment]
Currently it isn't 😞 

The problem here is that in the implementation the fragment doesn't actually get the full object. So as far as the the fragment is concerned, there is no way to reach the "parent" element. 

I also have to admit that I'm currently no very actively improving on grunt-bake. 

But needless to say, I always appreciate valuable contributions.

**Comment by MathiasPaumgarten** [comment]
Alright, so I might be wrong about this actually because I haven't used it myself this way. (Ha, that's a weird thing to say about ones own code)

In addition to setting a section, you could also pass an additional parameter that is just called `globalMeta="metas"` and that would pass the global meta section.

**Comment by hiulit** [comment]
Well, it's a pity that you're not maintaining grunt-bake, because I've used it, and I'm using it, in a lot of projects. I really like it! 😄  So, first of all, a big thank you for all you've already done! 🎉 

Now, I've tried your solution and it doesn't seem to work properly.

I have this:
```
<!--(bake /includes/header.html _section="pages.home" globalMeta="{{metas}}")-->
```

In the `header.html` I added a couple of tests:

```
{{globalMeta}} -> This prints [object Object], so it seems like it's working.
{{globalMeta.author}} -> This prints nothing, and it should print "Sir Arthur Conan Doyle" from "metas" if I'm correct.
```

From here, I don't know what else to do.
Could you point me in the right direction? Did I miss something there?

And what about the fragment implementation? I could try and improve (If you "guide" me).

Thanks a lot for your time!

**Comment by MathiasPaumgarten** [comment]
I'm so sorry for the crazy delay. 
I must not be evaluating variable like this. Honestly I forgot how I evaluate regular inline variables. And I do know that I worked with some others on this as well. 

I can't give you a straight forward answer, if you're willing to look into this it would start somewhere around there: [Line 467](https://github.com/MathiasPaumgarten/grunt-bake/blob/master/tasks/bake.js#L467)

**Comment by hiulit** [comment]
Don't worry about the delay :)

Good news! I think I managed to make this issue work.

[Line 495](https://github.com/MathiasPaumgarten/grunt-bake/blob/master/tasks/bake.js#L495)

```js
// resolve placeholders within inline values so these can be used in subsequent grunt-tags (see #67)
inlineValues = mout.object.map( inlineValues, function( value ) {
    return processContent( value, values );
} );
```

```js
// resolve placeholders within inline values so these can be used in subsequent grunt-tags (see #67)
inlineValues = mout.object.map( inlineValues, function( value ) {
    if ( mout.string.contains( value, '{' ) && mout.string.contains( value, '}' ) ) {
        value = mout.string.trim( value, [ '{', '}' ] );
        value = mout.object.get( parentValues, value );
        return value;
    } else {
        return processContent( value, values );
    }
} );
```
Basically what I did here is:

- Check if `value` has `{` and `}`.
- If so, remove them to get a clean `value`.
- Use the resulting `value` to get an `Object`.
- Return the `value` as an `Object`.

I know that the `parsePattern` can be altered by the user, so hardcoding the `{` and `}` it's a bad idea, but I don't know how to check it dynamically.

**Comment by MathiasPaumgarten** [comment]
You know, that is actually already pretty nice of a solution. I think I can clean it up and use the `parsePattern` or at least a somewhat cleaner regex, but you kind of did the heavy lifting on this already. 

I'll try to get this in as soon as I can.

**Comment by hiulit** [comment]
Awesome! 👍 
Again, if you guide me, I can happily try to clean it up and create a pull request.
Anyhow, what I did it's kinda working, so I can wait for you, no pressure at all!
Glad I could help :)

**Comment by MathiasPaumgarten** [comment]
Alright. So I played around a little bit and I found a little bit of a problem with the cleanest possible solution. The thing is: Inline values are already being parsed, but if you put in an object, like I suggested, it will actually convert it to a string. this is why `globalMetas` in 
```<!--(bake /includes/header.html _section="pages.home" globalMeta="{{metas}}")-->```
left you with `[object Object]`. Because it will understand that you pass in an object but turn that into a string. This was a features discussed a long time ago in #67 

This could be resolved by finding a specific syntax that does not convert it to an object.

Essentially in your solution you avoid processing alltogether and do exactly that. Which isn't bad but a more general solution would maybe be even nicer. Such as defining a raw version of the `{{ variable }}` syntax. Perhpas something like `{{! variable }}`.

**Comment by MathiasPaumgarten** [comment]
I wrote simple fix for this [here](https://github.com/MathiasPaumgarten/grunt-bake/blob/feature-inline-values-objects/tasks/bake.js#L480)

Still have to clean it up and document it. But I'm pretty sure I can push this soon. Would not be a breaking change in any way.

**Comment by MathiasPaumgarten** [comment]
I've implemented it over here #103 

Closing this.

**Comment by hiulit** [comment]
Hey, sorry for the super late reply...

It works perfectly! Thank you! 👍
